### ISSUE TITLE
[MotS] Where's an astromech when you need one?

### ISSUE THREAD

**Issue by Vortikai** [shinyquagsire23/OpenJKDF2#178]
While testing for #157, the astromech droid fails to appear (gets stuck) preventing the player from progressing through a locked door that leads to level completion. I am 95% sure this is a vanilla issue but could we attempt to fix it once and for all?

https://user-images.githubusercontent.com/63983128/211803609-adeaff2b-d445-4d84-8556-c035676c3e9b.mp4

**Comment by shinyquagsire23** [comment]
I think sometimes what happens is an enemy can spawn inside the room and trap the droid, wasn't able to get it stuck but I did get an enemy to randomly appear in there with the droid

**Comment by Vortikai** [comment]
It's mega RNG. For 20 years I'd say it happened to me 10% of the time. I think the droid gets stuck with an enemy as you say or gets frightened somehow.

**Comment by Vortikai** [comment]
A quick idea is that whenever Mara activates that panel to force open the door and/or reset the astromech.

**Comment by Alex3474247** [comment]
It's ok in my system.
![image](https://user-images.githubusercontent.com/17110303/219875294-1354fd02-1a25-4fb6-a8c1-8f008a1036b4.png)

**Comment by Vortikai** [comment]
This is an RNG LEC bug.

**Comment by shimstock22** [comment]
> This is an RNG LEC bug.

Yes, the related cog was a disaster.  I honestly don't know how it ever worked in the first place.  With my testing, it actually worked far far less often than it should have.  So, this was pretty major to me and felt like it deserved a patch from LEC.  It is fixed in my patched level set. Stay tuned for its release.
